packages/core/src/service.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  BindingFilter,
+  Constructor,
+  ContextView,
+  inject,
+  InjectionMetadata,
+  MetadataInspector,
+  transformValueOrPromise,
+} from '@loopback/context';
+
+/**
+ * Metadata for `@service` injection
+ */
+export interface ServiceInjectionMetadata extends InjectionMetadata {
+  /**
+   * Do not check sub classes to match service bindings
+   */
+  skipSubClasses?: boolean;
+}
+
+/**
+ * `@service` injects a service with matching class
+ * @param serviceClass - Constructor for the service. If not provided, the value
+ * is inferred from the design:type of the parameter or property
+ *
+ * @example
+ * ```ts
+ * export class MyController {
+ *   constructor(@service(MyService) private myService: MyService) {}
+ *
+ *   @service()
+ *   private logger: Logger;
+ * }
+ * ```
+ */
+export function service(
+  serviceClass?: Constructor<unknown>,
+  metadata?: ServiceInjectionMetadata,
+) {
+  return inject(
+    '',
+    {decorator: '@service', ...metadata},
+    (ctx, injection, session) => {
+      let serviceType: Function | undefined = serviceClass;
+      if (!serviceType) {
+        if (typeof injection.methodDescriptorOrParameterIndex === 'number') {
+          serviceType = MetadataInspector.getDesignTypeForMethod(
+            injection.target,
+            injection.member!,
+          ).parameterTypes[injection.methodDescriptorOrParameterIndex];
+        } else {
+          serviceType = MetadataInspector.getDesignTypeForProperty(
```

Comment:
This PR looks good to me. 
I just have a question since I am not that familiar with `injection`. 
The case that `injection.methodDescriptorOrParameterIndex === 'number'` is for when the decorator is applied to a method parameter. 
How about `MetadataInspector.getDesignTypeForProperty`? Does it mean that decorator is applied to a method property? Could you explain a bit? thanks!